test/test_torch.py

```diff
@@ -17923,6 +17922,46 @@ def is_view_of(self, base, other):
 
         return True
 
+    @onlyOnCPUAndCUDA
+    def test_views_with_dtype_change(self, device):
```

Comment:
this is just testing view_as_real and view_as_complex right? 
maybe split this in two tests since you don't reuse code?